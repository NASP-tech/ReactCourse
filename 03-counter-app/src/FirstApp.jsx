import PropTypes from 'prop-types';

// const newMessage = [1,2,3,4,5,6,7,8,9];
const newMessage = {
    message: 'Hola Mundo',
    title: 'Natalia',
};

const getResult = () => {
    return 4 + 4;
}

const getSum = (a, b) => {
    return a + b;
}

const FirstApp = ({
    title, 
    subtitle,
    name
}) => {   
    
    if(!title){
        throw new Error('El title no existe');
    }

    return (
        <>
            <code>{JSON.stringify(newMessage)}</code>
            {/* <h1>{ 1 + 1 }</h1> */}
            <h1>{getResult()}</h1>
            <h1>{getSum(2,3)}</h1>
            <h1>{title}</h1>
            <h1>{subtitle + 1}</h1>
            <h1>{name}</h1>
            {/* <p>Lorem, ipsum dolor.</p> */}
            {/** Para hacer comentarios*/}
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 'No hay subtitulo',
    name: 'Natalia'
}

export default FirstApp;