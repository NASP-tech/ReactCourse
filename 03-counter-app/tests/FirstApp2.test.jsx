import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe('Pruebas en <FirstApp/>', () => {

    const title = 'Hola, soy Natalia';
    const subtitle = 'Hola, soy un subtitulo';

    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>)
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje "Hola, soy Natalia"', () => { 
        render(<FirstApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
         //screen.debug();
    })

    test('Debe de mostrar el titulo en un h1', () => { 
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)        
    })

    test('Debe de mostrar el subtitulo enviado por props', () => {
        const { getAllByText } = render(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
});