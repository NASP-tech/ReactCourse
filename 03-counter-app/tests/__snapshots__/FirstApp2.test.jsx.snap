// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Natalia
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Natalia
  </p>
</div>
`;
